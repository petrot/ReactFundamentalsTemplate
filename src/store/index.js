import { applyMiddleware, configureStore } from "@reduxjs/toolkit";
import userSlice from "./slices/userSlice";
import coursesSlice from "./slices/coursesSlice";
import authorsSlice from "./slices/authorsSlice";
import thunk from "redux-thunk";

const store = configureStore(
  {
    reducer: {
      user: userSlice,
      courses: coursesSlice,
      authors: authorsSlice,
    },
  },
  applyMiddleware(thunk)
);

export default store;
