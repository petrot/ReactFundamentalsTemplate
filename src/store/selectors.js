// Module 3:
// * create selectors

export const getCoursesSelector = (state) => state.courses;
export const getAuthorsSelector = (state) => state.authors;
export const getUserSelector = (state) => state.user;
export const getUserNameSelector = (state) => state.user?.name;
export const getUserRoleSelector = (state) => null;
export const getUserTokenSelector = (state) => null;
