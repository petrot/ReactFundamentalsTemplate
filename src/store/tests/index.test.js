import store from "../index";

describe("Store", () => {
  it("should have been initialized", async () => {
    expect(store).toMatchSnapshot();
  });
});
