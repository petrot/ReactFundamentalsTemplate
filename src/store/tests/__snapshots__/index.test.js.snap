// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store should have been initialized 1`] = `
Object {
  "@@observable": [Function],
  "dispatch": [Function],
  "getState": [Function],
  "replaceReducer": [Function],
  "subscribe": [Function],
}
`;
